janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -509,16 +529,18 @@ public List<RequestBytes> next() {
                 if (requestSerializedSize + chunkSerializedTotal <= bulkChunkSerializedLimitBytes) {
                     chunkSerializedTotal += requestSerializedSize;
                     serializedRequests.add(requestIterator.next());
-                } else if (requestSerializedSize > bulkChunkSerializedLimitBytes) {
-                    //we've encountered an element we cannot send to Elasticsearch given the configured limit
-                    throw new IllegalArgumentException(String.format(
-                        "Bulk request item is larger than permitted chunk limit. Limit is %s. Serialized item size was %s",
-                        bulkChunkSerializedLimitBytes, requestSerializedSize));
                 } else {
                     //Adding this element would exceed the limit, so return the chunk
                     return serializedRequests;
                 }
```

Comment:
Seems we can avoid duplicate arithmetical operation: 
```
                chunkSerializedTotal += peeked.getSerializedSize();
                if (chunkSerializedTotal <= bulkChunkSerializedLimitBytes) {
                    serializedRequests.add(requestIterator.next());
                } else {
                    //Adding this element would exceed the limit, so return the chunk
                    return serializedRequests;
                }
```

Reply:
Changed in the latest force push